projects/habitat_ovmm/eval_dataset.py

```diff
@@ -49,31 +50,49 @@ def create_ovmm_env_fn(config):
     return env
 
 
-class VectorizedEvaluator(PPOTrainer):
-    """Class for creating vectorized environments, evaluating OpenVocabManipAgent on an episode dataset and returning metrics"""
+def process_and_adjust_config(args: Namespace) -> DictConfig:
+    """
+    Process and adjust the configuration based on the provided arguments.
 
-    def __init__(self, config, config_str: str):
-        self.visualize = config.VISUALIZE or config.PRINT_IMAGES
-
-        if not self.visualize:
-            # TODO: not seeing any speed improvements when removing these sensors
-            if "robot_third_rgb" in config.habitat.gym.obs_keys:
-                config.habitat.gym.obs_keys.remove("robot_third_rgb")
-            if (
-                "third_rgb_sensor"
-                in config.habitat.simulator.agents.main_agent.sim_sensors
-            ):
-                config.habitat.simulator.agents.main_agent.sim_sensors.pop(
-                    "third_rgb_sensor", None
-                )
+    This function takes a Namespace object containing parsed command-line arguments and performs the following steps:
+    1. Merges the habitat and baseline configurations.
+    2. Removes third person sensors to improve speed if visualization is not required.
+    3. Processes the episode range if specified and updates the EXP_NAME accordingly.
+
+    Args:
+        args (Namespace): The parsed command-line arguments.
+
+    Returns:
+        DictConfig: The processed and adjusted configuration.
+    """
 
-        episode_ids_range = config.habitat.dataset.episode_indices_range
-        if episode_ids_range is not None:
-            config.EXP_NAME = os.path.join(
-                config.EXP_NAME, f"{episode_ids_range[0]}_{episode_ids_range[1]}"
+    config, _ = get_config(args.habitat_config_path, opts=args.opts)
+    baseline_config = OmegaConf.load(args.baseline_config_path)
+    config = DictConfig({**config, **baseline_config})
+    visualize = config.VISUALIZE or config.PRINT_IMAGES
+
+    if not visualize:
+        # TODO: not seeing any speed improvements when removing these sensors
+        if "robot_third_rgb" in config.habitat.gym.obs_keys:
+            config.habitat.gym.obs_keys.remove("robot_third_rgb")
+        if "third_rgb_sensor" in config.habitat.simulator.agents.main_agent.sim_sensors:
+            config.habitat.simulator.agents.main_agent.sim_sensors.pop(
+                "third_rgb_sensor", None
             )
-        OmegaConf.set_readonly(config, True)
 
+    episode_ids_range = config.habitat.dataset.episode_indices_range
+    if episode_ids_range is not None:
+        config.EXP_NAME = os.path.join(
+            config.EXP_NAME, f"{episode_ids_range[0]}_{episode_ids_range[1]}"
+        )
+    OmegaConf.set_readonly(config, True)
+    return config
+
+
+class VectorizedEvaluator(PPOTrainer):
```

Comment:
Can we move VectorizedEvaluator class to the separate file (for example, `vectorized_evaluator.py`) and then import it in the `eval_dataset.py`?

After we move VectorizedEvaluator to the separate file can we also rename `eval_dataset.py` to `eval_ovmm_agent.py` or better `eval_baseline_ovmm_agent.py`? 

Then we can define Random agent in `eval_random_ovmm_agent.py` (similar to how it is done in [habitat-challenge](https://github.com/facebookresearch/habitat-challenge/blob/main/agents/agent.py)).

Reply:
Updated. Please check.

Reply:
@yvsriram LGTM, evaluation code runs on my machine. Thank you!

Reply:
Thanks for testing.